libtransmission/peer-common.h

```diff
@@ -171,7 +172,7 @@ class tr_peer_event
     }
 };
 
-using tr_peer_callback = void (*)(tr_peer* peer, tr_peer_event const& event, void* client_data);
+using tr_peer_callback = std::function<void(tr_swarm&, tr_peer*, tr_peer_event const&)>;
```

Comment:
I like the idea of moving to a std::function but would rather not add the tr_swarm reference to the signature; I may be reusing this callback and would rather not couple it to tr_swarm.

Could you change the tr_peer_callback signature so that it's not coupled to tr_swarm?

Reply:
Hm... hopefully yes, but this comes with a cost of some redesigning of `tr_peer` and related bits.